### PR TITLE
Clarify how retention interacts with readers

### PR DESCRIPTION
**Pull request by grantwwu**
See Slack conversation:

https://apache-pulsar.slack.com/archives/C5Z4T36F7/p1563815652030700

### CODE REVIEW

**Comment by merlimat** [comment]
> See Slack conversation: 

Since Slack text is not visible without an account, and that it gets deleted after a while, it might be good to copy&paste the text here rather than linking

File: site2/docs/cookbooks-retention-expiry.md [hunk 1]
```diff
@@ -4,34 +4,40 @@ title: Message retention and expiry
...
+By default, when a Pulsar message arrives at a broker it will be stored until it has been acknowledged on all subscriptions, at which point it will be marked for deletion. You can override this behavior and retain even messages that have already been acknowledged on all subscriptions by setting a *retention policy* for all topics in a given namespace. Retention policies are either a *size limit* or a *time limit*.
+
+Retention policies are particularly useful if you intend to exclusively use the Reader interface. Because the Reader interface does not use acknowledgements, messages will never exist within backlogs. Most realistic Reader-only use cases require that retention be configured.
 
 When you set a size limit of, say, 10 gigabytes, then messages in all topics in the namespace, *even acknowledged messages*, will be retained until the size limit for the topic is reached; if you set a time limit of, say, 1 day, then messages for all topics in the namespace will be retained for 24 hours.
 
-It is also possible to set *infinite* retention time or size, by setting `-1` for either time or
-size retention.
+TODO: Confirm this behavior?
```
Comment (grantwwu) [inline site2/docs/cookbooks-retention-expiry.md]: @merlimat can you confirm that this description is accurate?

File: site2/docs/cookbooks-retention-expiry.md [hunk 2]
```diff
@@ -122,6 +128,8 @@ admin.namespaces().getRetention(namespace);
 
 You can control the allowable size of backlogs, at the namespace level, using *backlog quotas*. Setting a backlog quota involves setting:
 
+TODO: Expand on is this per backlog or per topic?
```
Comment (grantwwu) [inline site2/docs/cookbooks-retention-expiry.md]: And if this is per topic, how is it computed?  (See Slack conversation).

File: site2/docs/cookbooks-retention-expiry.md (hunk 1 above)
Comment (sijie) [reply site2/docs/cookbooks-retention-expiry.md]: yes  -1 is for infinite retention.

File: site2/docs/cookbooks-retention-expiry.md (hunk 2 above)
Comment (sijie) [reply site2/docs/cookbooks-retention-expiry.md]: this is per topic.

**Comment by sijie** [comment]
run java8 tests

File: site2/docs/cookbooks-retention-expiry.md (hunk 1 above)
Comment (grantwwu) [reply site2/docs/cookbooks-retention-expiry.md]: Oops, sorry, wrong line.

File: site2/docs/cookbooks-retention-expiry.md (hunk 2 above)
Comment (grantwwu) [reply site2/docs/cookbooks-retention-expiry.md]: Er, so to clarify:

Suppose I have 10 subscriptions on a topic.  They have not acknowledged the last 1000 messages, which are 1 kilobyte each.

Is the value that is being compared to the quota 1 megabyte or 10 megabytes?

Comment (sijie) [reply site2/docs/cookbooks-retention-expiry.md]: the backlog is the number of messages. so the value is 1000.

**Comment by grantwwu** [comment]
Er, this got merged with TODOs still in it...

**Comment by wolfstudy** [comment]
Change the Milestone to 2.4.2, because of conflict.

**Pull request merged**